### PR TITLE
Add route to count adherent by referent area

### PR DESCRIPTION
**Pull request by B-Galati**
https://app.clubhouse.io/enmarche/story/3552/1ère-partie-section-adhérents-graph-double-histogramme-adherents-membres-de-comités

### CODE REVIEW

File: src/History/CommitteeMembershipHistoryHandler.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    private $historyRepository;
+
+    public function __construct(CommitteeMembershipHistoryRepository $historyRepository)
+    {
+        $this->historyRepository = $historyRepository;
+    }
+
+    public function queryCountByMonth(Adherent $referent, int $months = 6): array
+    {
+        foreach (range(0, $months - 1) as $monthInterval) {
```
Comment (ottaviano) [inline src/History/CommitteeMembershipHistoryHandler.php]: as an alternative:
```
$startDate = (new Chronos('last day of -6 month'))->setTime(23, 59, 59, 999);
$endDate = new Chronos();

$period = new \DatePeriod($startDate, new \DateInterval('P1M'), $endDate);

foreach ($period as $date) {
    // $date -> each month
}
```

File: src/Repository/CommitteeMembershipHistoryRepository.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            throw new \InvalidArgumentException('Adherent must be a referent.');
+        }
+
+        $query = $this->createQueryBuilder('history')
+            ->select('history.action, history.adherentUuid, COUNT(history) AS count')
+            ->join('history.referentTags', 'tags')
+            ->where('tags IN (:tags)')
+            ->andWhere('history.date <= :until')
+            ->groupBy('history.action, history.adherentUuid')
+
```
Comment (ottaviano) [inline src/Repository/CommitteeMembershipHistoryRepository.php]: CS: extra line

File: tests/Committee/CommitteeManagerTest.php [hunk 3]
```diff
@@ -7,6 +7,7 @@
 use AppBundle\DataFixtures\ORM\LoadAdherentData;
 use AppBundle\DataFixtures\ORM\LoadCommitteeMembershipHistoryData;
 use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Committee;
```
Comment (ottaviano) [inline tests/Committee/CommitteeManagerTest.php]: unused, no?

File: src/History/CommitteeMembershipHistoryHandler.php (hunk 1 above)
Comment (B-Galati) [reply src/History/CommitteeMembershipHistoryHandler.php]: I like that !

File: tests/Committee/CommitteeManagerTest.php (hunk 3 above)
Comment (B-Galati) [reply tests/Committee/CommitteeManagerTest.php]: Yea good catch

File: src/History/CommitteeMembershipHistoryHandler.php (hunk 1 above)
Comment (B-Galati) [reply src/History/CommitteeMembershipHistoryHandler.php]: Sadly it does not work as expected :-(

image

image

Comment (B-Galati) [reply src/History/CommitteeMembershipHistoryHandler.php]: easier to read https://3v4l.org/C0JbP

Comment (ottaviano) [reply src/History/CommitteeMembershipHistoryHandler.php]: 😢 👍

File: src/Repository/CommitteeMembershipHistoryRepository.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+class CommitteeMembershipHistoryRepository extends ServiceEntityRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, CommitteeMembershipHistory::class);
+    }
+
+    public function countAdherentMemberOfAtLeastOneCommitteeManagedBy(Adherent $referent, \DateTimeInterface $until): int
+    {
+        if (!$referent->isReferent()) {
```
Comment (illusionOfParadise) [inline src/Repository/CommitteeMembershipHistoryRepository.php]: You can use a function `checkReferent` of this trait https://github.com/EnMarche/en-marche.fr/blob/master/src/Repository/ReferentTrait.php#L9

**Pull request merged**